tests/tt_metal/tools/tt_builder/test_builder.cpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <filesystem>
+#include "tools/tt_builder/builder.hpp"
+#include "llrt/hal.hpp"
+#include "tt_metal/common/logger.hpp"
+
+TEST(BuilderTest, test_firmware_build) {
+    tt::tt_metal::tt_builder builder;
+
+    std::filesystem::path output_path("/tmp/test_builder_firmware_build/");
```

Comment:
Is it a bad idea for the test to generate binary files into this path `/tmp/test_builder_firmware_build/` ? What would be a better place while running this test in CI pipelines? @tt-rkim 
FYI : I haven't enabled this test in pipeline yet, will do before merging this PR.